tests/acceptance/forms/general/NonInlineFormCest.php

```diff
@@ -0,0 +1,455 @@
+<?php
+/**
+ * Tests for non-inline ConvertKit Forms.
+ *
+ * @since   2.3.9
+ */
+class NonInlineFormCest
+{
+	/**
+	 * Run common actions before running the test functions in this class.
+	 *
+	 * @since   2.3.9
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function _before(AcceptanceTester $I)
+	{
+		// Activate ConvertKit plugin.
+		$I->activateConvertKitPlugin($I);
+	}
+
+	/**
+	 * Test that no forms are displayed for the Default Form (Site Wide) option
+	 * when no non-inline forms exist.
+	 *
+	 * @since   2.3.9
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function testSettingsWhenNoNonInlineForms(AcceptanceTester $I)
+	{
+		// Setup Plugin with API Keys for an account that has no non-inline forms.
+		$I->setupConvertKitPlugin($I, $_ENV['CONVERTKIT_API_KEY_NO_DATA'], $_ENV['CONVERTKIT_API_SECRET_NO_DATA']);
+	}
+
+	/**
+	 * Test that the defined default non-inline form displays site wide.
+	 *
+	 * @since   2.3.9
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function testDefaultNonInlineForm(AcceptanceTester $I)
+	{
+		// Setup Plugin with a non-inline Default Form (Site Wide).
+		$I->setupConvertKitPlugin(
```

Comment:
For these setup functions with tons of parameters (and this goes for non-setup functions too), it's hard to read what this is doing. I wonder if using one big array so you can name the parameters you're passing. I know that might not be the style, but I would have to have `setupConvertKitPlugin` definition right next to this call to understand it. I know that PHP 8 introduced named arguments which is the best solution for this, but maybe an array could tide us over. Not necessarily in this PR, just wanted to provide some feedback on this.